addons/binding/org.openhab.binding.smartthings/src/main/java/org/openhab/binding/smartthings/internal/SmartthingsHttpClient.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.smartthings.internal;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.eclipse.jetty.http.HttpMethod;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * HTTP client used to send messages to smartthings hub
+ *
+ * @author Bob Raker - Initial contribution
+ *
+ */
+public class SmartthingsHttpClient {
+
+    private final Logger logger = LoggerFactory.getLogger(SmartthingsHttpClient.class);
+
+    private HttpClient httpClient;
+    private Gson gson;
+
+    private String smartthingsIp;
+    private int smartthingsPort;
+
+    public SmartthingsHttpClient(String smartthingsIp, int smartthingsPort) throws Exception {
+        // Save the Smartthing IP and port
+        this.smartthingsIp = smartthingsIp;
+        this.smartthingsPort = smartthingsPort;
+
+        // Setup the jetty client for use in sending data to the Smartthings hub
+        httpClient = new HttpClient();
```

Comment:
That's not allowed as per coding guidelines.
You must instead "reference" the OSGi service `HttpClientFactory` (in your factory/discovery component or other components) and get a client from there. The reason is that the framework want a single ssl configuration and control about open ports.